consumption.tex

```diff
@@ -0,0 +1,74 @@
+
+\begin{itemize}
+  \item \textsf{tx}: The transaction which contains the burn on the source blockchain.
+  \item \textsf{b}: The block where the aforementioned transaction is contained.
+  \item $\tau$: An inclusion proof showing that $\mathsf{tx} \in \mathsf{b}$ such that $\VerTxIncl(\textsf{b}, \textsf{tx}, \tau) = \true$.
+  \item \textsf{blockConnection}: A proof that \textsf{b} is contained in the best (i.e., most proof-of-work) source blockchain.
+\end{itemize}
+
+Putting these together, the final proof is the tuple $\beta = \sf(tx, \tau, b, blockConnection)$.
+
+Transferring value from a source to a target blockchain has already been a subject of research. A mechanism for the so-called 1-way pegs is described in~\cite{pow-sidechains}. We adapt a modified version of this scheme for our purposes, to achieve our own 1-way peg by making use of Proof-of-Burn. A 1-way peg is established when there is the ability to transfer an amount of source cryptocurrency to a target blockchain in a trustless manner, as we wish to do. Generally, the prover wishes to convince the verifier that an event has taken place on the source blockchain. We define an event as a simple value transfer described by (a) a transaction id \textsf{txid}, (b) a receiving address \textsf{receiver} and (c) an amount \textsf{amount}. Thus our event takes the form of the tuple $e = (\textsf{txid}, \textsf{receivingAddress}, \textsf{amount})$. We remark that simple value transfers are supported by all popular cryptocurrencies, allowing a verifier to process burns from a wide range of source blockchains. We also remark that for now this event type does not yet distinguish between burn and non-burn addresses.
+
+This verifier contract is responsible for receiving, verifying and then storing events on the other blockchain. It relies on a function $\mathsf{verify}^{e,\mathcal{G}}(\mathcal{P})$, where $\mathcal{P}$ is a set of proofs for the event, which returns $\true$ if and only if the event is valid based on the set of proofs provided. The functions of the protocol, in the order in which they should be called are as follows:
+
+\begin{itemize}
+  \item $\textsf{submit-event-proof}(\pi, e)$: A prover calls this function while putting down a collateral $z$ with an event $e$ and a proof of it $\pi$. The function determines if the proof provided is valid by making use of the \textsf{verify} method. If the proof is valid, a contestation period of $k$ blocks begins where other provers may submit a contesting proof.
+  \item $\textsf{submit-contesting-proof}(\pi^*, e)$: A prover calls this function if they have a contesting proof for the proof provided previously for event $e$. If through the use of \textsf{verify} it is discovered that the original prover was dishonest, her collateral gets transferred to the caller of this function and the event is discarded.
```

Comment:
This protocol already deviates from the federation implementation, where there are no contesting proofs. I think we can leave all of this discussion for the decentralized consumption subsection.

Reply:
I have changed the flow, mentioning the decentralized approach first and the federation afterwards